src/RequestHandle.cpp

```diff
@@ -401,6 +470,19 @@ auto RequestHandle::onComplete() -> void {
     m_on_complete_handler(std::move(request));
 }
 
+
+auto RequestHandle::getBytesLeftToWrite() -> size_t
+{
+    if (m_max_download_bytes > -1)
```

Comment:
do you want to do a bounds check and make sure m_bytes_written is not greater than m_max_download_bytes? otherwise you'll underflow (if there is a bug somewhere) and return a very large number

Reply:
Good point! Added bounds checking and it returns an optional -- if false, we have downloaded too many bytes, otherwise it should be true and have the difference.